multiqc/modules/biscuit/biscuit.py

```diff
@@ -0,0 +1,688 @@
+        self.general_stats_addcols(pd, pheader)
+
+        # if len(mdata)>0 and all([dd['Duplication Rate'] is not None for dd in mdata.values()]):
+        # self.general_stats_addcols(mdata, {'% Dup': {'suffix':'%', 'min':0, 'max':100}})
+
+
+    def parse_logs_mapq(self, f, fn):
+
+        data = {}
+        if fn.endswith('_mapq_table.txt'):
+            for l in f.splitlines():
+                s = l.split()
+                data[s[0]] = s[1]   # mapping quality > number of reads
+                
+        if fn.endswith('_strand_table.txt'):
+            for l in f.splitlines():
+                m = re.search(r'strand\t([+-]*)\t(\d+)', l)
+                if m is not None:
+                    data[m.group(1)] = int(m.group(2))
+
+        if fn.endswith('_isize_score_table.txt'):
+            data['I'] = {}      # insert size
+            data['S'] = {}      # SW-score
+
+            for l in f.splitlines():
+                fields = l.split()
+                if fields[0] == 'I':
+                    data[fields[0]][int(fields[1])] = float(fields[2]) * 100
+                elif fields[0] == 'S':
+                    data[fields[0]][fields[1]] = float(fields[2])
+                
+        return data
+    
+    def chart_biscuit_mapq(self):
+
+        # fraction of optimally mapped reads
+        pd = {}
+        for sid, dd in self.mdata['mapq'].items():
+            if not sid.endswith('_mapq_table.txt'):
+                continue
+            sid = sid.replace('_mapq_table.txt','');
+            pd[sid] = {'OAligned':0, 'SAligned':0, 'UAligned':1}
+            for mapq, cnt in dd.items():
+                if mapq == 'unmapped':
+                    pd[sid]['UAligned'] += int(cnt)
+                elif int(mapq) >= 40:
+                    pd[sid]['OAligned'] += int(cnt)
+                else:
+                    pd[sid]['SAligned'] += int(cnt)
+
+        self.add_section(
+            name = 'Mapping Summary',
+            anchor = 'biscuit-mapping',
+            plot = bargraph.plot(pd, OrderedDict([
+                ('OAligned', {'name':'Optimally Aligned Reads', 'color': '#8bbc21'}),
+                ('SAligned', {'name':'Suboptimally Aligned Reads', 'color': '#f7a35c'}),
+                ('UAligned', {'name':'Unaligned Reads', 'color': '#000000'})
+            ]), {'id':'biscuit_mapping_summary',
+                 'title':'BISCUIT: Mapping Summary',
+                 'ylab':'Number of Reads',
+                 'cpswitch_c_active': True,
+                 'cpswitch_counts_label': '# Reads'
+            })
+        )
+
+        # mapping strand distribution
+        pd = dict([(k.replace('_strand_table.txt', ''),v) for k,v in self.mdata['mapq'].items() if k.endswith('_strand_table.txt')])
+        self.add_section(
+            name='Mapping Strand Distribution',
+            anchor='biscuit-strands',
+            description = "<p>This plot shows the distribution of strand of mapping and strand of bisulfite conversion.</p>",
```

Comment:
The description is parsed as markdown, so no need for HTML tags such as `<p>`.

It would be great if you could write a `helptext` talking about different possibilities here and what to expect. eg. different types of bisulfite library prep.

Reply:
"<p>" are removed. helptext is added.